Backend/DynamicQR/DynamicQR.Infrastructure/Services/QrCodeRepositoryService.cs

```diff
@@ -65,14 +63,12 @@ public async Task<bool> UpdateAsync(string organisationId, QrCode qrCode, Cancel
 
             if (response.IsError)
                 throw new StorageException(response.ReasonPhrase);
-
-            return true;
         }
-
-        throw new StorageException();
+        else
+            throw new StorageException();
```

Comment:
Deze else en throw kunnen weg, als er correct geüpdatet is dan mag de code gewoon doorgaan.

Reply:
Ik denk dat er een andere error moet komen, dat het item niet gevonden kan worden. Anders weet je niet dat het niet gelukt is

Reply:
Je hebt gelijk, doordat de code was ingeklapt las ik het verkeerd. Ik las dit:
```
if (response.IsError) 
   throw new StorageException(response.ReasonPhrase);
else
   throw new StorageException();
```

In dat geval is een Exception gooien wel degelijk een goed idee. Ik zou het zo doen:

```
if (qrcodeFount == null)
   throw new ItemNotFoundException() // <-- Custom Exception
```

En dan die ook opvangen en daar een HttpStatusCode.404 Not found van maken.